packages/console/src/components/MultiTextInput/index.tsx

```diff
@@ -0,0 +1,254 @@
+import React, { useMemo, useRef, useState } from 'react';
+import {
+  useController,
+  Control,
+  FieldPath,
+  UnpackNestedValue,
+  PathValue,
+  Path,
+  FieldValues,
+  Primitive,
+} from 'react-hook-form';
+import { IsTuple, TupleKeys, ArrayKey } from 'react-hook-form/dist/types/path/common';
+import { useTranslation } from 'react-i18next';
+
+import * as textButtonStyles from '@/components/TextButton/index.module.scss';
+import Minus from '@/icons/Minus';
+
+import IconButton from '../IconButton';
+import TextInput from '../TextInput';
+import * as styles from './index.module.scss';
+
+type StringArrayPathImpl<K extends string | number, V> = V extends Primitive
+  ? never
+  : V extends ReadonlyArray<infer U>
+  ? U extends string
+    ? `${K}`
+    : never
+  : `${K}.${StringArrayPath<V>}`;
+
+type StringArrayPath<T> = T extends ReadonlyArray<infer V>
+  ? IsTuple<T> extends true
+    ? {
+        [K in TupleKeys<T>]-?: StringArrayPathImpl<K & string, T[K]>;
+      }[TupleKeys<T>]
+    : StringArrayPathImpl<ArrayKey, V>
+  : {
+      [K in keyof T]-?: StringArrayPathImpl<K & string, T[K]>;
+    }[keyof T];
+
+type StringFiledArrayPath<TFieldValues extends FieldValues> = StringArrayPath<TFieldValues>;
+
+type InputsRule = {
+  pattern: RegExp;
+  message: string;
+};
+
+type MutiTextInputErrors = {
+  inputs: Record<number, string | undefined>;
+  required: string | undefined;
+};
+
+type Props<
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends StringFiledArrayPath<TFieldValues> = StringFiledArrayPath<TFieldValues>
```

Comment:
`Filed` or `Field`?

Reply:
Align with the naming of react-hook-form.

Reply:
updated